instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/struts2/Struts2ActionSpanTest.java

```diff
@@ -0,0 +1,186 @@
+    try {
+      // struts 2.3
+      strutsFilterClass =
+          Class.forName("org.apache.struts2.dispatcher.ng.filter.StrutsPrepareAndExecuteFilter");
+    } catch (ClassNotFoundException exception) {
+      // struts 2.5
+      strutsFilterClass =
+          Class.forName("org.apache.struts2.dispatcher.filter.StrutsPrepareAndExecuteFilter");
+    }
+    context.addFilter(
+        (Class<? extends Filter>) strutsFilterClass, "/*", EnumSet.of(DispatcherType.REQUEST));
+
+    server.start();
+    return server;
+  }
+
+  @Override
+  protected void stopServer(Server server) throws Exception {
+    server.stop();
+    server.destroy();
+  }
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    options.setContextPath("/context");
+    options.setTestPathParam(true);
+    options.setTestErrorBody(false);
+    options.setHasHandlerSpan(endpoint -> !endpoint.equals(NOT_FOUND));
+    options.setHasResponseSpan(
+        endpoint ->
+            endpoint == REDIRECT
+                || endpoint == ERROR
+                || endpoint == EXCEPTION
+                || endpoint == NOT_FOUND);
+
+    options.setExpectedHttpRoute(
+        (ServerEndpoint endpoint, String method) -> {
+          if (method.equals(HttpConstants._OTHER)) {
+            return getContextPath() + endpoint.getPath();
+          }
+          if (endpoint.equals(PATH_PARAM)) {
+            return getContextPath() + "/path/{id}/param";
+          } else if (endpoint.equals(NOT_FOUND)) {
+            return getContextPath() + "/*";
+          } else {
+            return super.expectedHttpRoute(endpoint, method);
+          }
+        });
+  }
+
+  @Override
+  protected SpanDataAssert assertResponseSpan(
+      SpanDataAssert span, SpanData parentSpan, String method, ServerEndpoint endpoint) {
+    if (endpoint.equals(REDIRECT)) {
+      span.satisfies(spanData -> assertThat(spanData.getName()).endsWith(".sendRedirect"));
+    } else if (endpoint.equals(NOT_FOUND)) {
+      span.satisfies(spanData -> assertThat(spanData.getName()).endsWith(".sendError"))
+          .hasParent(parentSpan);
+    }
+
+    span.hasKind(SpanKind.INTERNAL);
+    return span;
+  }
+
+  @Override
+  protected SpanDataAssert assertHandlerSpan(
+      SpanDataAssert span, String method, ServerEndpoint endpoint) {
+    span.hasName("GreetingAction." + endpoint.name().toLowerCase(Locale.ROOT))
+        .hasKind(SpanKind.INTERNAL);
+
+    if (endpoint.equals(EXCEPTION)) {
+      span.hasStatus(StatusData.error())
+          .hasEventsSatisfyingExactly(
+              event ->
+                  event
+                      .hasName("exception")
+                      .hasAttributesSatisfyingExactly(
+                          equalTo(EXCEPTION_TYPE, Exception.class.getName()),
+                          equalTo(EXCEPTION_MESSAGE, EXCEPTION.getBody()),
+                          satisfies(EXCEPTION_STACKTRACE, val -> val.isInstanceOf(String.class))));
```

Comment:
I think you could replace this with `hasException(new Exception(EXCEPTION.getBody()))`